modules/042-kube-dns/monitoring/prometheus-rules/kubernetes/dns.yaml

```diff
@@ -15,3 +15,25 @@
         CoreDNS pod {{$labels.pod}} has at least one critical error.
         To debug the problem, look into container logs: `kubectl -n kube-system logs {{$labels.pod}}`
       summary: CoreDNS has critical errors.
+
+- name: deckhouse.kube-dns.legacy-services
+  rules:
+  - alert: D8KubeDnsServiceWithDeprecatedAnnotation
+    expr: d8_kube_dns_deprecated_service_annotation == 1
+    for: 5m
+    labels:
+      severity_level: "7"
+    annotations:
+      plk_markup_format: markdown
+      plk_protocol_version: "1"
+      plk_create_group_if_not_exists__d8_kube_dns_deprecated_resources: D8KubeDnsDeprecatedResources,tier=~tier,d8_module=kube-dns,d8_component=kube-dns
+      plk_grouped_by__d8_kube_dns_deprecated_resources: D8KubeDnsDeprecatedResources,tier=~tier,d8_module=kube-dns,d8_component=kube-dns
+      summary: Deprecated Service annotation found.
+      description: |
+        Replace deprecated Service annotation `service.alpha.kubernetes.io/tolerate-unready-endpoints` with `spec.publishNotReadyAddresses: true` ASAP.
```

Comment:
```suggestion
        Replace deprecated Service annotation `service.alpha.kubernetes.io/tolerate-unready-endpoints` with `spec.publishNotReadyAddresses: true`.
```

Reply:
I've been trying to emphasize the gravity of the situation. I've reworded it a bit.